compiler/luci/service/src/CircleShapeInferenceRule.cpp

```diff
@@ -1676,6 +1676,39 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
     return loco::NodeShape{output_shape};
   }
 
+  loco::NodeShape visit(const luci::CircleSparseToDense *node) final
+  {
+    loco::TensorShape shape;
+    {
+      LUCI_ASSERT(node->output_shape(), "dims input should not be nullptr");
+
+      auto output_shape_node = dynamic_cast<luci::CircleConst *>(node->output_shape());
+      if (output_shape_node != nullptr)
+      {
+        // Only support node with S32
+        LUCI_ASSERT(output_shape_node->dtype() == loco::DataType::S32,
+                    "Only support int32 CircleConst");
+
+        if (output_shape_node->rank() != 1)
+          INTERNAL_EXN_V("Only support rank 1 CircleConst",
+                         oops::to_uint32(output_shape_node->rank()));
+
+        shape.rank(output_shape_node->dim(0).value());
+
+        for (uint32_t axis = 0; axis < shape.rank(); ++axis)
```

Comment:
Is `output_shape_node->rank()` always 1 or current implementation only supports for 1 and later we may support for rank() > 1 ?
If this is for former case, we won't need to run a for loop. If second case, may be add `TODO support rank() > 1` so that loop will be used in the future...

Reply:
In implementation of TF V1.14, output_shape_node always rank 1.

Reply:
Ah... I didn't read the code correctly: `shape.rank(output_shape_node->dim(0).value());`. My mistake.
Code looks OK :)